xarray/core/merge.py

```diff
@@ -547,6 +547,20 @@ def dataset_merge_method(dataset, other, overwrite_vars, compat, join):
 
 
 def dataset_update_method(dataset, other):
-    """Guts of the Dataset.update method"""
+    """Guts of the Dataset.update method
+
+    This drops a duplicated coordinates from `other` (GH:2068)
+    """
+    from .dataset import Dataset
+    from .dataarray import DataArray
+
+    for k, obj in other.items():
+        if isinstance(obj, (Dataset, DataArray)):
+            # drop duplicated coordinates
+            coord_names = [k for k in obj.coords
+                           if k not in obj.dims and k in dataset.coords]
+            if coord_names:
+                other[k] = obj.drop(*coord_names)
```

Comment:
This will potentially modify the `other` argument in-place, which we try to avoid (e.g., `ds.update(other)` could modify `other`, not just `ds`).

I think it would work to make a shallow copy of `other` first? e.g., `other = other.copy()` works on both dictionaries and xarray.Dataset objects.

Reply:
Oops. You are right. Updated.